### ISSUE TITLE
Use Filename as Testsuite name?

### ISSUE THREAD

**Issue by dangoslen** [palmerj3/jest-junit#29]
Had an idea for a feature that I wanted to talk about before just trying to code something.

One of the disconnects between how this library and how JUnit seems to typically export test results is around the name of the testsuite. It seems that testsuites from typical maven executions (failsafe, surefire) use the filename as the name of the testsuite. Jest-junit is using either the descriptor assigned in the `describe(...)` code blocks or the file path depending on the configuration. This makes our jest-junit reports look very different than our other JUnit test reports

It would be great to be able to specify either the default behavior (using the `describe(...)` descriptor, using the file path or using the file name. An example might be:

```
"jest-junit" : {
     "suiteNameTemplate": "{filename|filepath|ancestorTitles}"  
}
```

This name would be similar to the other possible configuration values such as `classNameTemplate` and `titleTemplate`. 

The biggest catch here would be to understand the behavior of using this proposed idea with the existing `usePathForSuiteName` property. Maybe allow this property to override that existing value if set for a certain release number until potentially deprecating?

Let me know your thoughts!

**Comment by palmerj3** [comment]
Hello again!

Ok I want to better understand what you're proposing here.

Right now on a sample project with two test files and the following configuration:
```
  "jest-junit": {
    "suiteName": "jest tests",
    "output": "./junit.xml",
    "classNameTemplate": "{classname}-{title}",
    "titleTemplate": "{classname}-{title}",
    "ancestorSeparator": " › ",
    "usePathForSuiteName": "true"
  }
```

I get this:
```
<testsuites name="jest tests">
  <testsuite name="/__tests__/bar.test.js" tests="1" errors="0" failures="0" skipped="0" timestamp="2017-10-10T21:56:49" time="0.323">
    <testcase classname="bar-should be bar" name="bar-should be bar" time="0.004">
    </testcase>
  </testsuite>
  <testsuite name="/__tests__/foo.test.js" tests="1" errors="0" failures="0" skipped="0" timestamp="2017-10-10T21:56:49" time="0.322">
    <testcase classname="foo-should be foo" name="foo-should be foo" time="0.004">
    </testcase>
  </testsuite>
</testsuites>
```

Which contains the path to the file.

Did you have something else in mind?

**Comment by dangoslen** [comment]
Ahh. Should have included a concrete example.

The core idea is to use the same 'tag' idea present in the other template config options in this library for use in the name of each test suite. These tags would be the `filename`, `filepath` and `ancestorTitles`. The `filename` would just be the name of the file that generated tests, the `filepath` would be the full path to the file (including the filename) and `ancestorTitles` would be the values provided by jest from the `describe()` blocks

For instance, if I were to update the config in your example above to use the proposed idea:
```
"jest-junit": {
    "suiteName": "jest tests",
    "output": "./junit.xml",
    "classNameTemplate": "{classname}-{title}",
    "titleTemplate": "{classname}-{title}",
    "ancestorSeparator": " › ",
    "suiteNameTemplate": {filename}"
  }
```
the output would be:
```
<testsuites name="jest tests">
  <testsuite name="bar.test.js" tests="1" errors="0" failures="0" skipped="0" timestamp="2017-10-10T21:56:49" time="0.323">
    <testcase classname="bar-should be bar" name="bar-should be bar" time="0.004">
    </testcase>
  </testsuite>
  <testsuite name="foo.test.js" tests="1" errors="0" failures="0" skipped="0" timestamp="2017-10-10T21:56:49" time="0.322">
    <testcase classname="foo-should be foo" name="foo-should be foo" time="0.004">
    </testcase>
  </testsuite>
</testsuites>
```

Another concrete example would be
```
"jest-junit": {
    "suiteName": "jest tests",
    "output": "./junit.xml",
    "classNameTemplate": "{classname}-{title}",
    "titleTemplate": "{classname}-{title}",
    "ancestorSeparator": " › ",
    "suiteNameTemplate": {filepath}"
  }
```
generates
```
<testsuites name="jest tests">
  <testsuite name="/__tests__/bar.test.js" tests="1" errors="0" failures="0" skipped="0" timestamp="2017-10-10T21:56:49" time="0.323">
    <testcase classname="bar-should be bar" name="bar-should be bar" time="0.004">
    </testcase>
  </testsuite>
  <testsuite name="/__tests__/foo.test.js" tests="1" errors="0" failures="0" skipped="0" timestamp="2017-10-10T21:56:49" time="0.322">
    <testcase classname="foo-should be foo" name="foo-should be foo" time="0.004">
    </testcase>
  </testsuite>
</testsuites>
```
which is the same behavior as your original example.

Is that helping to clear things up a bit?

**Comment by palmerj3** [comment]
@dangoslen ahh ok - yes that's quite clear now. I love it!

Feel free to submit a PR and I'll review

**Comment by dangoslen** [comment]
Hey Jason,

Hope I'm not being annoying, but do you think you will have some time to update the deployed package to npm with these changes? Not sure the process there, so apologies if I missed something in my initial PR. Would like to get it to the teams I work with at Bandwidth for some of our projects.

Thanks!

**Comment by palmerj3** [comment]
@dangoslen sorry for the delay! I usually publish right away after a PR is merged. That's my fault. I just published 3.2.1 which includes your PR.

Let me know if there are any issues!

**Comment by dangoslen** [comment]
No worries at all. Just wanted to move away from pointing teams to using the github repo and just use npm. :)
